Observer/RecommendSettings.php

```diff
@@ -143,9 +147,10 @@ protected function validateRecommendation(string $changedPath, string $recommend
     {
         try {
             $recommendations = $this->recommendManagement->$recommendationMethod($this->getProductId());
-            if (empty($recommendations['renderingContent'])) {
-                throw new LocalizedException(__(
-                    "It appears that there is no trained model available for Algolia application ID %1.",
+            if ($this->shouldDisplayWarning($recommendations)) {
+                $this->messageManager->addWarningMessage(__(
```

Comment:
It's way better like this, I guess the models can take a while to be trained so displaying a warning instead of an error makes definitely more sense.